### PR TITLE
Flav/core function calls history anthropic

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

See https://github.com/dust-tt/dust/issues/4968.

This PR applies the same logic as https://github.com/dust-tt/dust/pull/5029 for Anthropic. It adds support for `function_calls` and their results in the message history.

## Risk

Worst case it breaks the chat block for Anthropic. Safe to rollback.

### CODE REVIEW

File: core/src/providers/anthropic.rs [hunk 1]
```diff
@@ -1185,25 +1269,6 @@ impl LLM for AnthropicLLM {
             },
         );
 
-        messages = messages
```
Comment (flvndvd) [inline core/src/providers/anthropic.rs:1188]: Hey @spolu, do you recall the reason behind always using a single item in the content Vec? This was introduced in https://github.com/dust-tt/dust/pull/4180.

**Review by fontanierh — APPROVED** [review]
LGTM 🙏

File: core/src/providers/anthropic.rs (hunk 1 above)
Comment (spolu) [reply core/src/providers/anthropic.rs:1188]: No good reason IMHO.

Did you double check the code below (we need to keep it around for now due to old assistant conversation rendering) but just want to make sure it won't interfere too much ?

Comment (spolu) [reply core/src/providers/anthropic.rs:1188]: Sorry looks like it's the code above :)

Since you updated the comment. Looks good

**Review by spolu — APPROVED** [review]

File: core/src/providers/anthropic.rs (hunk 1 above)
Comment (flvndvd) [reply core/src/providers/anthropic.rs:1188]: Yep from my tests no regression are introduced by removing this logic.

**Pull request merged**